Project-Aurora/Project-Aurora/Settings/VariableRegistry.cs

```diff
@@ -20,21 +24,66 @@ public class VariableRegistryItem
         public string Remark = "";
         public VariableFlags Flags = VariableFlags.None;
 
-        public VariableRegistryItem()
+        [JsonIgnore]
+        public IBindable Bindable { get; set; }
+
+        public VariableRegistryItem() { }
+
+        public VariableRegistryItem(IBindable bindable, string title = "", string remark = "", VariableFlags flags = VariableFlags.None)
         {
+            Bindable = bindable;
+            switch (Bindable)
+            {
+                case Bindable<bool> b:
+                    Value = b.Value;
+                    Default = b.Default;
+                    break;
+                case BindableNumber<double> b:
+                    Value = b.Value;
+                    Default = b.Default;
+                    Max = b.MaxValue;
+                    Min = b.MinValue;
+                    break;
+                case BindableNumber<float> b:
+                    Value = b.Value;
+                    Default = b.Default;
+                    Max = b.MaxValue;
+                    Min = b.MinValue;
+                    break;
+                case BindableNumber<int> b:
+                    Value = b.Value;
+                    Default = b.Default;
+                    Max = b.MaxValue;
+                    Min = b.MinValue;
+                    break;
+                case BindableNumber<long> b:
+                    Value = b.Value;
+                    Default = b.Default;
+                    Max = b.MaxValue;
+                    Min = b.MinValue;
+                    break;
+                case BindableColor b:
+                    Value = b.Value;
+                    Default = b.Default;
+                    break;
+                case Bindable<string> b:
+                    Value = b.Value;
+                    Default = b.Default;
+                    break;
+            }
+            Title = title;
+            Remark = remark;
+            Flags = flags;
         }
```

Comment:
What if I want to add a new bindable type? Do I have to add it in here too? This has the potential to become another massive chunk of code. And what about the other BindableNumbers you have, such as short?

Reply:
A tiny little miss yesterday when i changed most of the switches over to Bindable<object> as i was really tired when i made that change.